src/xtgeo/surface/regular_surface.py

```diff
@@ -225,145 +310,69 @@ def __init__(
-                (ncol, nrow), C order will be made. When used with a file-like input,
-                values can be a bool, where False will indicate that values shall
-                not be loaded, only metadata.
+                (ncol, nrow), C order will be made.
             masked: Indicating if numpy array shall be masked or not. Default is True.
             name: A given name for the surface, default is file name root or
-                'unkown' if constructed from scratch.
+                'unknown' if constructed from scratch.
 
         Examples:
-            The instance can be made either from file or by a spesification::
-
-                >>> x1 = RegularSurface('somefilename')  # assume Irap binary
-                >>> x2 = RegularSurface('somefilename', fformat='irap_ascii')
-                >>> x3 = RegularSurface().from_file('somefilename',
-                                                    fformat='irap_binary')
-                >>> x4 = RegularSurface()
-                >>> x4.from_file('somefilename', fformat='irap_binary')
-                >>> x5 = RegularSurface(ncol=20, nrow=10, xori=2000.0, yori=2000.0,
-                                        rotation=0.0, xinc=25.0, yinc=25.0,
-                                        values=np.zeros((20,10)))
+            The instance can be made by specification::
+
+                >>> surface = RegularSurface(
+                    ncol=20,
+                    nrow=10,
+                    xori=2000.0,
+                    yori=2000.0,
+                    rotation=0.0,
+                    xinc=25.0,
+                    yinc=25.0,
+                    values=np.zeros((20,10))
+                    )
+
 
         """
-        #
         logger.info("Start __init__ method for RegularSurface object %s", id(self))
-        self._ncol = 5
-        self._nrow = 3
-        self._xori = 0.0
-        self._yori = 0.0
-        self._xinc = 25.0
-        self._yinc = 25.0
-        self._rotation = 0.0
-
-        self._undef = xtgeo.UNDEF
-        self._undef_limit = xtgeo.UNDEF_LIMIT
-
-        self._masked = False
-        self._filesrc = None  # Name of original input file or stream, if any
-        self._name = "unknown"  # informal name for use in plots etc
+        self._ncol = ncol
+        self._nrow = nrow
+        self._xori = xori
+        self._yori = yori
+        self._xinc = xinc
+        self._yinc = yinc
+        self._rotation = rotation
+        self._yflip = yflip
+        self._name = name
 
-        # These are useful when import/export of surfaces with seismic origin
-        self._ilines = None
-        self._xlines = None
+        self._undef = undef
+        self._undef_limit = xtgeo.UNDEF_LIMIT
 
-        self._yflip = 1
+        self._filesrc = filesrc  # Name of original input file or stream, if any
 
-        self._values = None
         self._fformat = None  # current fileformat, useful for load()
```

Comment:
The `fformat` is consumed and discarded in the factory loader, hence this will always be `None`. But the `load` function could use this later though:

```
if not self._isloaded:
    self.from_file(self._filesrc, fformat=self._fformat)
    self._isloaded = True
```

I don't think this should stay in its current shape, and if deemed critical we will have to include the `fformat` in the constructor again

Reply:
How do you suggest we deal with it? Should we include `kwargs` in the constructor?

Reply:
I wouldn't argue for including `kwargs` in the constructor. Typical reason to use `kwargs` is avoiding to have to modify signature on methods that eventually will call other functions, hence extensibility. Issue with `kwargs` is obfuscating what `kwargs` could be - and the docstring _will_ become outdated and one needs to go through the code instead..

I would rather include the `fformat` in the constructor itself, as long as we support using lazy loading it is a viable use of an argument.

Or would you favour `kwargs`?